libs/core/uagents_core/utils/registration.py

```diff
@@ -0,0 +1,229 @@
+import hashlib
+import json
+
+import requests
+from typing import Optional, Union, List, Dict
+from pydantic import BaseModel, Field
+import time
+import urllib.parse
+
+from uagents_core.crypto import Identity
+from uagents_core.types import AgentEndpoint, AgentType
+from uagents_core.config import (
+    DEFAULT_AGENTVERSE_URL,
+    DEFAULT_ALMANAC_API_PATH,
+    DEFAULT_MAILBOX_API_PATH,
+    DEFAULT_CHALLENGE_PATH,
+)
+from uagents_core.logger import get_logger
+
+
+
+
+logger = get_logger("uagents_core.utils.registration")
+
+
+class VerifiableModel(BaseModel):
+    agent_address: str
```

Comment:
This will likely need an update very soon based on this PR: https://github.com/fetchai/uAgents/pull/603.